### PR TITLE
[gcs/ha] Fix java related error when enable redisless ray

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
This PR enables ray java to be able to run without redis. It also fixes java related tests and updated the pipeline.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by iycheng** [comment]
This PR depends on this https://github.com/ray-project/ray/pull/21628
I'll merge to master once the upstream PR merged.

**Review by mwtian — APPROVED** [review]

**Review by WangTaoTheTonic — APPROVED** [review]
Looks totally fine to me!

**Comment by iycheng** [comment]
- Lint also failed in master.

**Pull request merged**
